controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1405,6 +1416,143 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	return actionUpdate{steadyStateResult}
 }
 
+// DataImage handler for attaching/detaching image.
+func (r *BareMetalHostReconciler) handleDataImageActions(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	dataImage := &metal3api.DataImage{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, dataImage); err != nil {
+		// DataImage does not exist or it may have been deleted
+		if k8serrors.IsNotFound(err) {
+			info.log.Info("dataImage not found")
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		return actionError{errors.Wrap(err, "could not load dataImage")}
+	}
+
+	// Set ControllerReference to DataImage
+	if !ownerReferenceExists(info.host, dataImage) {
+		if err := controllerutil.SetControllerReference(info.host, dataImage, r.Scheme()); err != nil {
+			return actionError{errors.Wrap(err, "could not set bmh as controller")}
+		}
+		if err := r.Update(info.ctx, dataImage); err != nil {
+			return actionError{errors.Wrap(err, "failure creating dataImage resource")}
+		}
+		// Should we requeue at this point
+		// return actionContinue{}
+	}
+
+	// Fetch the latest status of DataImage from node
+	if err := prov.GetDataImageStatus(dataImage); err != nil {
+		info.log.Info("Failed to get current status : ", "ErrorIs", err)
```

Comment:
nit: Please remove " : ", there is no guarantee how exactly the fields are logged (also s/ErrorIs/Error/ for the same reason)